docs/design/in-doc-search-ui.rst

```diff
@@ -0,0 +1,183 @@
+but we need to decide that which option would be best for our use-case.
+
+Edge NGram Tokenizer
+~~~~~~~~~~~~~~~~~~~~
+
+* Pros
+
+  * More effective than Completion Suggester when it comes to autocompleting
+    words that can appear in any order.
+  * It is considerable fast because most of the work is being done at index time,
+    hence the time taken for autocompletion is reduced.
+  * Supports highlighting of the matching terms.
+
+* Cons
+
+  * Requires greater disk space.
+
+
+Completion Suggester
+~~~~~~~~~~~~~~~~~~~~
+
+* Pros
+
+  * Really fast as it is optimized for speed.
+  * Does not require large disk space.
+
+* Cons
+
+  * Matching always starts at the beginning of the text. So, for example,
+    "Hel" will match "Hello, World" but not "World Hello".
+  * Highlighting of the matching words is not supported.
+  * According to the official docs for Completion Suggester,
+    fast lookups are costly to build and are stored in-memory.
+
+
+Milestones
+----------
+
++-----------------------------------------------------------------------------------+------------------+
+| Milestone                                                                         | Due Date         |
++===================================================================================+==================+
+| A local implementation of the project.                                            | 12th June, 2019  |
++-----------------------------------------------------------------------------------+------------------+
+| In-doc search on a test project hosted on Read the Docs using the RTD Search API. | 20th June, 2019  |
++-----------------------------------------------------------------------------------+------------------+
+| In-doc search on docs.readthedocs.io.                                             | 20th June, 2019  |
++-----------------------------------------------------------------------------------+------------------+
+| Friendly user trial where users can add this on their own docs.                   | 5th July, 2019   |
++-----------------------------------------------------------------------------------+------------------+
+| Additional UX testing on the top-10 Sphinx themes.                                | 15th July, 2019  |
++-----------------------------------------------------------------------------------+------------------+
+| Finalize the UI.                                                                  | 25th July, 2019  |
++-----------------------------------------------------------------------------------+------------------+
+| Improve the search backend for efficient and fast search results.                 | 10th August, 2019|
++-----------------------------------------------------------------------------------+------------------+
+
+
+Open Questions
+++++++++++++++
+
+* Should we rely on jQuery, any third party library or pure vanilla JavaScript?
+* Are the subprojects to be searched?
```

Comment:
Yes, but the backend API should already be doing this.

Reply:
Yes. It is a question when we are changing/updating the backend.